### PR TITLE
`rustc` tool analysis: add detectable column

### PR DESCRIPTION
**Pull request by Urhengulas**
I am uncertain if detectability is set correct for all potential errors. Please review carefully.

My thinking was that everything input related can be reviewed since it is clear what options there are and is therefore detectable. Everything internal or output related is not detectable, because it is unclear what should be emitted/generated.

### CODE REVIEW

File: ferrocene/doc/evaluation-report/src/rustc/tool-analysis.rst [hunk 1]
```diff
@@ -37,58 +39,68 @@ Rust Driver
...
    * - .. id:: RUSTC_ERR_DRIVER_09
      - :id:`RUSTC_UC1_RLIB`, :id:`RUSTC_UC2_STATICLIB`, :id:`RUSTC_UC3_EXEC`, :id:`RUSTC_UC4_EXEC_RLIB`, :id:`RUSTC_UC5_EXEC_CLIB`
      - An incomplete input is accepted leading to an undefined behavior
      - Undefined behavior
      - :id:`RUSTC_AVD_TEST_007`
+     - YES
    * - .. id:: RUSTC_ERR_DRIVER_10
      - :id:`RUSTC_UC1_RLIB`, :id:`RUSTC_UC2_STATICLIB`, :id:`RUSTC_UC3_EXEC`, :id:`RUSTC_UC4_EXEC_RLIB`, :id:`RUSTC_UC5_EXEC_CLIB`
-     - Some object files are not generated silently
+     - Some object files are silently not generated
```
Comment (Urhengulas) [inline ferrocene/doc/evaluation-report/src/rustc/tool-analysis.rst:100]: The old wording sounds wrong to me, or was it correct?

File: ferrocene/doc/evaluation-report/src/rustc/tool-analysis.rst [hunk 2]
```diff
@@ -193,43 +219,50 @@ Linking
...
      - Description
      - Risk
      - Mitigation
+     - Detectable
    * - .. id:: RUSTC_ERR_LINK_23
      - :id:`RUSTC_UC3_EXEC`, :id:`RUSTC_UC4_EXEC_RLIB`, :id:`RUSTC_UC5_EXEC_CLIB`
      - Invalid input is accepted
      - Undefined behavior
      - :id:`RUSTC_AVD_CHECK_BUILD_SCRIPT_003`
+     - YES
```
Comment (pietroalbini) [inline ferrocene/doc/evaluation-report/src/rustc/tool-analysis.rst:235]: Whether the linker accepts invalid input is not easy to detect.

File: ferrocene/doc/evaluation-report/src/rustc/tool-analysis.rst [hunk 3]
```diff
@@ -146,43 +165,50 @@ LLVM
...
      - Description
      - Risk
      - Mitigation
+     - Detectable
    * - .. id:: RUSTC_ERR_LLVM_17
      - :id:`RUSTC_UC1_RLIB`, :id:`RUSTC_UC2_STATICLIB`, :id:`RUSTC_UC3_EXEC`, :id:`RUSTC_UC4_EXEC_RLIB`, :id:`RUSTC_UC5_EXEC_CLIB`
      - Input parameter has invalid value
      - Most likely LLVM will crash. Invalid code could also be generated
      - :id:`RUSTC_AVD_TEST_007`
+     - YES
```
[the remark this answers is not in the document]
Comment (Urhengulas) [reply ferrocene/doc/evaluation-report/src/rustc/tool-analysis.rst:181]:
```suggestion
     - NO
```

File: ferrocene/doc/evaluation-report/src/rustc/tool-analysis.rst (hunk 2 above)
Comment (Urhengulas) [reply ferrocene/doc/evaluation-report/src/rustc/tool-analysis.rst:235]:
```suggestion
     - NO
```

**Comment by tshepang** [comment]
looks good... please squash

**Review by tshepang — APPROVED** [review]
bors merge

**Pull request merged**